driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/vault/ClientEncryptionImpl.java

```diff
@@ -132,7 +135,7 @@ public Mono<BsonBinary> createDataKey(final String kmsProvider, final DataKeyOpt
                             .getDatabase(namespace.getDatabaseName())
                             .getCollection(namespace.getCollectionName(), BsonDocument.class)
                             .withWriteConcern(WriteConcern.MAJORITY);
-                    return Mono.from(collectionWithTimeout(vaultCollection, operationTimeout) //TODO should be defered?
+                    return Mono.from(collectionWithTimeout(vaultCollection, operationTimeout, TIMEOUT_ERROR_MESSAGE_CREATE_DATA_KEY)
```

Comment:
Removing the TODO - means it didn't need to be a deferred mono?

Reply:
Yes, because` flatMap` triggers a subscription immediately after the Mono is returned from the lambda, the execution timing of the deferred operation remains effectively the same as if a regular `Mono` were returned.